crates/engine/tree/src/tree/precompile_cache.rs

```diff
@@ -0,0 +1,188 @@
+//! Contains a precompile cache that is backed by a moka cache.
+
+use alloy_primitives::Bytes;
+use reth_evm::precompiles::{DynPrecompile, Precompile};
+use revm::precompile::{PrecompileError, PrecompileResult};
+use std::sync::Arc;
+
+/// Cache for precompiles, for each input stores the result.
+pub type PrecompileCache =
+    mini_moka::sync::Cache<CacheKey, CacheEntry, alloy_primitives::map::DefaultHashBuilder>;
+
+/// Create a new [`PrecompileCache`].
+pub fn create_precompile_cache() -> PrecompileCache {
+    mini_moka::sync::CacheBuilder::new(100_000)
+        .build_with_hasher(alloy_primitives::map::DefaultHashBuilder::default())
+}
```

Comment:
I'd prefer this as a standalone `PrecompileCache(Arc<>)` type instead, because I find it easier to trace access to internals like that, and we could get rid of the outer `Arc<PrecompileCache>`

and could introduce additional functions in case we need them

Reply:
very good point, done